engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -4021,10 +4021,23 @@ public NicProfile addVmToNetwork(final VirtualMachine vm, final Network network,
         }
     }
 
+    /**
+     * duplicated in {@see UserVmManagerImpl} for a {@see UserVmVO}
+     */
+    private void checkIfNetExistsForVM(VirtualMachine virtualMachine, Network network) {
+        List<NicVO> allNics = _nicsDao.listByVmId(virtualMachine.getId());
```

Comment:
Do we need an additional check for the nics state?

Reply:
not sure what you mean, there may be only one nic per network per VM. Does the state matter, @nvazquez ?